urlcdd0c1ab49_emiln_JHouseFireExplorer_tgz-pJ8-dk_brics_automaton_DatatypesJ8/src/Util.java

```diff
@@ -0,0 +1,108 @@
+import java.io.*;
+import java.util.*;
+import java.awt.*;
+import dk.brics.automaton.*;
+
+public class Util {
+    public static boolean isPossible4chanAutomaton(Automaton aut) {
+        try {
+            RegExp reg = new RegExp("(.|\".\"|\"/\"){8}");
+            Automaton legal = reg.toAutomaton();
+            return !aut.intersection(legal).isEmpty();
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+    }
+    public static boolean isValidRegExp(String regExp) {
+        try {
+            RegExp re = new RegExp(regExp);
+        } catch (IllegalArgumentException e) {
+            return false;
+        }
+        return true;
+    }
+    public static void writeTextToFile(String text, String path) {
+        try {
+            FileWriter fw = new FileWriter(path);
+            BufferedWriter bw = new BufferedWriter(fw);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *bw*. The resource is referenced at lines 27, 28. The resource is closed at line 29. There are other execution paths that do not contain closure statements, for example, when Writer.write throws an exception. Ensure *bw* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).